File: sol/2009.cpp
#include <algorithm>

struct Solution {
    int minOperations(vector<int>& nums) {
        size_t size = nums.size(), j = 0;
        size_t ans = size;
        sort(nums.begin(), nums.end());
        nums.erase(unique(nums.begin(), nums.end()), nums.end());
        for (size_t i = 0; i < nums.size(); ++i){
            while (j < nums.size() && nums[j] < nums[i] + size){
                ++j;
            }
            ans = min(ans, size - j + i);
        }
        return ans;
    }
};

File: sol/2009.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        n = len(nums)
        nums = sorted(frozenset(nums))
        ans = n
        e = 0
        for i, x in enumerate(nums):
            while e < len(nums) and nums[e] < x + n:
                e += 1
            ans = min(ans, n - e + i)
        return ans
